math/mathcore/inc/Math/DistSampler.h

```diff
@@ -92,93 +92,118 @@ class DistSampler {
 
 
    /**
-       Set the random engine to be used
+       Set the random engine to be used.
        To be implemented by the derived classes who provides
        random sampling
    */
    virtual void SetRandom(TRandom *  ) {}
 
    /**
        Set the random seed for the TRandom instances used by the sampler
-       classes
+       classes.
        To be implemented by the derived classes who provides random sampling
    */
    virtual void SetSeed(unsigned int /*seed*/ ) {}
 
    /**
-      Get the random engine used by the sampler
+      Get the random engine used by the sampler.
       To be implemented by the derived classes who needs it
       Returns zero by default
     */
    virtual TRandom * GetRandom() { return 0; }
 
-   /// set range in a given dimension
+   /// Set the range in a given dimension.
    void SetRange(double xmin, double xmax, int icoord = 0);
 
-   /// set range for all dimensions
+   /// Set the range for all dimensions.
    void SetRange(const double * xmin, const double * xmax);
+   /// Set the range for all dimensions (use std::vector)
+   void SetRange(const std::vector<double> & xmin, const std::vector<double> & xmax){ 
+      assert(xmin.size() >= NDim() && xmax.size() >= NDim());
+      SetRange(xmin.data(),xmax.data());
+   }
 
-   /// set range using DataRange class
+   /// Set the range using the ROOT::Fit::DataRange class.
    void SetRange(const ROOT::Fit::DataRange & range);
 
-   /// set the mode of the distribution (could be useful to some methods)
-   /// implemented by derived classes if needed
+   /// Set the mode of the distribution (1D case).
+   /// It could be useful or needed by some sampling methods.
+   /// It is implemented by derived classes if needed (e.g. TUnuranSampler)
    virtual void SetMode(double  ) {}
 
-   /// set the normalization area of distribution
-   /// implemented by derived classes if needed
+   /// Set the mode of the distribution (Multi-dim case).
+   virtual void SetMode(const std::vector<double> &) {}
+
+   /// Set the normalization area of distribution.
+   /// Implemented by derived classes if needed
    virtual void SetArea(double) {}
 
-   /// get the parent distribution function (must be called after setting the function)
+   /// Use the log of the provided pdf.
+   /// Implemented by the derived classes 
+   virtual void SetUseLogPdf(bool = true) {}
+
+   /// Set usage of Derivative of PDF. 
+   /// Can be implemented by derived class 
+   virtual void SetDPdf(const ROOT::Math::IGenFunction & ) {}
+
+   /// Set usage of Cumulative of PDF.
+   /// Can be implemented by derived class
+   virtual void SetCdf(const ROOT::Math::IGenFunction &) {}
+
+   /// Get the parent distribution function (must be called after setting the function).
    const ROOT::Math::IMultiGenFunction & ParentPdf() const {
       return *fFunc;
    }
 
+   /// Check if there is a parent distribution defined.
+   bool HasParentPdf() const { return fFunc != nullptr; }
```

Comment:
@lmoneta Wouldn't `return fFunc;` be more elegant here?

Reply:
Personally I think the currently proposed `fFunc != nullptr` is more elegant because there is no implicit type conversion from pointer to bool.

Reply:
I added also a bool function becase there was already a ParentPdf() returning a reference, and it does not work whne fFunc is a nullptr